coalib/settings/ConfigurationGathering.py

```diff
@@ -122,6 +124,16 @@ def warn_config_absent(sections, argument, log_printer):
                          "to give the `--{}` argument?".format(argument))
 
 
+def configure_logging_from_json(filename):
+    """
+    Configures logging module based on parameters from JSON file.
+
+    :param filename:    Name of a file with logging configuration.
+    """
+    with open(filename) as cnf_file:
+        logging.config.dictConfig(json.load(cnf_file))
```

Comment:
Separation of concerns. When settings are separated, it's easier to find
and edit them. In particular, if you will want to add, say, the ability for
users to have their own logging settings, it will much easier to change
from configuration in the file. What are your thoughts? Do you think it'll
be better to have it somewhere in code?

On 20:22, Thu, Sep 8, 2016 Bence Nagy <notifications@github.com> wrote:

> In coalib/settings/ConfigurationGathering.py
> <https://github.com/coala-analyzer/coala/pull/2409#discussion_r78050580>:
>
> > @@ -122,6 +124,16 @@ def warn_config_absent(sections, argument, log_printer):
> >                           "to give the `--{}` argument?".format(argument))
> >
> >
> > +def configure_logging_from_json(filename):
> > +    """
> > +    Configures logging module based on parameters from JSON file.
> > +
> > +    :param filename:    Name of a file with logging configuration.
> > +    """
> > +    with open(filename) as cnf_file:
> > +        logging.config.dictConfig(json.load(cnf_file))
>
> What's the benefit of having a separate JSON file instead of just keeping
> the config somewhere as a Python dictionary?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/2409/files/7fb344c19abcc42926feeb3e74838f2f44e4cb8e..8c261cef9f1a652371a2c20d4a3d21ea30c165e4#r78050580>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADyQxUXZTO6pxD6W0kVaqm8ZJAFhbvhNks5qoERCgaJpZM4JB7lk>
> .
>